backend/Dockerfile

```
@@ -0,0 +1,78 @@
+ARG APP_UID=1001 # Valeur par défaut pour l'UID
+ARG APP_GID=1001 # Valeur par défaut pour le GID
+ARG APP_VERSION="unknown"
+ARG GIT_COMMIT="unknown"
+ARG BUILD_DATE="" # Ex: $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+
+# Création d'un groupe et d'un utilisateur dédiés non-root avec les UID/GID spécifiés.
+# Le '|| true' est une précaution si l'utilisateur/groupe existe déjà (peu probable avec --system dans une image fraîche).
+# Pour une meilleure idempotence, on peut vérifier l'existence avant d'ajouter/modifier.
+# Note: 'groupmod' et 'usermod' sont utilisés comme fallback si l'UID/GID est déjà pris par un utilisateur système.
+#       Dans des images de base minimales, il est plus simple d'assumer que l'UID/GID est libre.
+RUN groupadd --gid ${APP_GID} ${APP_GROUP} || groupmod -o -g ${APP_GID} ${APP_GROUP} && \
```

Hadolint (2.12.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)